### PR TITLE
utils: use container based cli

### PR DESCRIPTION
**Pull request by EmbeddedAndroid**
## Summary of Changes

The util scripts currently are using host binaries for running commands against devnet. This can be confusing to users as we also provide wrappers for accessing the same binaries running inside the container devenv.

Some examples are:

https://github.com/stacks-network/sbtc/blob/main/devenv/bitcoin/bin/bitcoin-cli

https://github.com/stacks-network/sbtc/blob/main/devenv/sbtc/bin/sbtc

Implements: https://github.com/stacks-network/sbtc/issues/224

This PR switches the utils scripts to use the wrappers for container binaries. For the binaries running inside devenv, it must use the service hostnames otherwise localhost will cause them to fail.

### Risks

The biggest risk I see is, if the sbtc container (romeo) crashes, the sbtc command: https://github.com/stacks-network/sbtc/blob/main/devenv/sbtc/bin/sbtc will not be able to run, as the container stopped.

### How were these changes tested?

Tested locally with devenv

### What future testing should occur?

We need to automate the entire flow via https://github.com/stacks-network/sbtc/pull/215

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by CAGS295 — APPROVED** [review]

File: devenv/utils/mine_btc.sh [hunk 1]
```diff
@@ -11,5 +11,4 @@ if [[ -z "$num_blocks" ]]; then
 fi
 
 btc_address=$(source $dir/get_credentials.sh | jq -r '.credentials["0"].bitcoin.p2wpkh.address')
-
-bitcoin-cli -rpcconnect=localhost -rpcport=18443 -rpcuser=devnet -rpcpassword=devnet generatetoaddress $num_blocks $btc_address
+$dir/../bitcoin/bin/bitcoin-cli generatetoaddress $num_blocks $btc_address
```
Comment (CAGS295) [inline devenv/utils/mine_btc.sh:14]: Why is it okay to drop the auth options?

**Review by friedger — APPROVED** [review]
Looks good

**Comment by friedger** [comment]
I just fixed the merge conflicts. Sorry, It is your task, I guess. And @CAGS295 needs to review again :-)

File: devenv/utils/mine_btc.sh [hunk 2]
```diff
@@ -12,4 +12,4 @@ fi
 
 btc_address=$(source $dir/get_credentials.sh | jq -r '.credentials["1"].bitcoin.p2wpkh.address')
 
-bitcoin-cli -rpcconnect=localhost -rpcport=18443 -rpcuser=devnet -rpcpassword=devnet generatetoaddress $num_blocks $btc_address
+bitcoin-cli -rpcconnect=localhost -rpcport=18443 -rpcuser=devnet -rpcpassword=devnet generatetoaddress $num_blocks $btc_address
```
Comment (CAGS295) [inline devenv/utils/mine_btc.sh:15]: Did we lose the options changes?

Comment (friedger) [reply devenv/utils/mine_btc.sh]: My fault :sweat:

**Comment by CAGS295** [comment]
@EmbeddedAndroid, how do you check that stacks-node is ready to deploy contracts? The heuristic I've used so far is checking for stacks height #2.

**Review by friedger — APPROVED** [review]
LGTM

**Review by friedger — APPROVED** [review]
Looks good

**Review by CAGS295 — APPROVED** [review]

**Pull request merged**